src/dreyfus_httpd.erl

```diff
@@ -190,22 +186,30 @@ analyze(Req, Analyzer, Text) ->
             send_error(Req, Reason)
     end.
 
-parse_index_params(#httpd{method='GET'}=Req) ->
+parse_index_params(#httpd{method='GET'}=Req, Db) ->
     IndexParams = lists:flatmap(fun({K, V}) -> parse_index_param(K, V) end,
         chttpd:qs(Req)),
-    parse_index_params(IndexParams);
-parse_index_params(#httpd{method='POST'}=Req) ->
+    parse_index_params(IndexParams, Db);
+parse_index_params(#httpd{method='POST'}=Req, Db) ->
     IndexParams = lists:flatmap(fun({K, V}) -> parse_json_index_param(K, V) end,
         element(1, chttpd:json_body_obj(Req))),
-    parse_index_params(IndexParams);
-parse_index_params(IndexParams) ->
-    Args = #index_query_args{},
+    parse_index_params(IndexParams, Db);
+parse_index_params(IndexParams, Db) ->
+    DefaultLimit = case fabric_util:is_partitioned(Db) of
+        true ->
+            list_to_integer(config:get("dreyfus", "limit_partitions", "2000"));
```

Comment:
So for non-partitions our default is 25, max is 200. For partitions, default is 2000, and max is also 2000? Just wanted to make sure that's intended.

Reply:
That'd be a question for @jiangphcn on what the requirements said about that.

Reply:
yes, there is no change on non-partition database. For partition database, the default and max are all 2000. https://github.ibm.com/cloudant/partition-queries/issues/44#issue-5513779